src/pybind/mgr/dashboard/frontend/src/app/shared/services/branding.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { ServicesModule } from './services.module';
+
+import { detect } from 'detect-browser';
+
+import { DashboardService } from '../api/dashboard.service';
+import { UserService } from '../api/user.service';
+
+@Injectable({
+  providedIn: ServicesModule
+})
+export class BrandingService {
+  product: any;
+
+  constructor(private dashboardService: DashboardService, private userService: UserService) {}
+
+  setContants() {
```

Comment:
Typo: missing 's' in ConStants. Nonetheless, I think it's a bit misleading to have a method called 'setConstants' (that can be called multiple times), given that constants are usually pre-set and never set multiple times. Additionally, some of these are not even constants (`data.mgr_map.services.dashboard` can change if a Ceph-Mgr failover happens).

IMHO, this service is a mixture of 2 different things:
- **String constants**: product name, company, version, license string, copyright notice, etc.
- **Environment details**: current Ceph-mgr hostname, user name, role, browser, ...

So my suggestion would be:
- Have a separate constants file for real constants, I [already dropped a few example links](https://stackoverflow.com/questions/34986922/define-global-constants-in-angular-2).
- Have a service (if you want) for getting environment details/info, but I wouldn't call it branding, as it has nothing to do with branding. 

I think both pieces could perfectly fit upstream (the environment info, and the global constants).

Reply:
ack

Reply:
@a2batic @epuertat 

I meant to ask about the About Modal.

The Host Address seems to show a URL.  It used to show the Ceph Manager host(s).  Is the Host Address showing the Ceph Manager API endpoint or other API end point?  If so, the label may need to change to reflect that, but let's determine what the intent was first.

Does the User and User Role now dynamically update?

Reply:
I think there are two possible sources for the IP address/hostname in the mgr_map (which is the source used for this): one is the current active Ceph-mgr hostname, and the other is the back-end service endpoint (a URL schema: http[s]://<ip>:<port>).

Reply:
@julienlim @epuertat, 'User' is updating as expected, 'User Role' API endpoint is only accessible to users with admin permissions, there are two possibilities: either we remove 'User Role' or show it for admins only.

Reply:
ack